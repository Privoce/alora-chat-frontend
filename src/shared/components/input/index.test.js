import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import {
	InputComponent
} from 'shared/components';

describe('<InputComponent />', () => {
	const onChange = sinon.spy();
	const onBlur = sinon.spy();

	const modelI = {
		id: 'nickname',
		placeholder: 'Nickname',
		type: 'text',
		value: 'value',
		onChange,
		onBlur,
		hasError: true,
		errorMessage: 'erro teste'
	};

	const modelII = {
		id: 'password',
		placeholder: 'password',
		type: 'password',
		value: 'value',
		onChange,
		onBlur,
		hasError: false
	};

	const getWrapper = (model) => {
		return shallow((
			<InputComponent {...model} />
		));
	};

	it('renders correctly', () => {
		const wrapper = getWrapper(modelI);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('has InputTextComponent component', () => {
		const wrapper = getWrapper(modelI);
		expect(wrapper.find('input').length).toBe(1);
	});

	it('has a error span when err param are true', () => {
		const wrapper = getWrapper(modelI);
		expect(wrapper.find('LabelComponent').length).toBe(1);
	});

	it('has no error span when err param are true', () => {
		const wrapper = getWrapper(modelII);
		expect(wrapper.find('span.error-label').length).toBe(0);
	});

	it('blur and change events are working', () => {
		const wrapper = getWrapper(modelI);
		wrapper.find('input').simulate('blur');
		wrapper.find('input').simulate('change');
		expect(onChange.called).toBe(true);
		expect(onBlur.called).toBe(true);
	});
});
