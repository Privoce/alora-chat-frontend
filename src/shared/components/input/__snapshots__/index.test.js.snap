// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent /> renders correctly 1`] = `
<div
  className="input-wrapper"
  style={
    Object {
      "margin": undefined,
      "maxWidth": undefined,
    }
  }
>
  <div
    className="input-container"
  >
    <input
      className="input input--error"
      id="nickname"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Nickname"
      style={
        Object {
          "height": undefined,
          "width": undefined,
        }
      }
      type="text"
      value="value"
    />
    <div
      className="input-icon-container"
    />
  </div>
  <div
    className="input-label-container"
  >
    <LabelComponent
      danger={true}
      fontRegular={true}
      fontSize={16}
      text="erro teste"
    />
  </div>
</div>
`;
